### PR TITLE
Look for synopsis factory up on INDEX startup

### PR DESCRIPTION
**Pull request by mavam**
When the INDEX starts up, it now sets a custom synopsis factory for the meta index when the runtime settings contain an entry for `Sy_FACTORY`, which in turn yields the key for a concrete factory function.

### CODE REVIEW

File: libvast/test/system/index.cpp [hunk 1]
```diff
@@ -199,3 +218,16 @@ TEST(iterable bro conn log query result) {
 }
 
 FIXTURE_SCOPE_END()
+
+FIXTURE_SCOPE(meta_index_setup_test, synopsis_fixture)
+
+TEST(meta index factory) {
+  auto factory = get_synopsis_factory(sys);
```
Comment (Neverlord) [inline libvast/test/system/index.cpp]: Any reason not to use `unbox(get_synopsis_factory(sys))`?

File: libvast/test/system/index.cpp [hunk 2]
```diff
@@ -199,3 +218,16 @@ TEST(iterable bro conn log query result) {
...
+  REQUIRE(factory);
+  CHECK_EQUAL(factory->first, caf::atom("Sy_TEST"));
+  CHECK(factory->second == make_synopsis);
+  // We could additionally go into the state of the actor and verify that the
```
Comment (Neverlord) [inline libvast/test/system/index.cpp]: We should absolutely verify the INDEX. It's also simple with CAF's testing DSL:

```c++
MESSAGE("run initialization code");
run();
MESSAGE("verify INDEX state");
auto& ref = deref<stateful_actor<index_state>>(index);
CHECK_EQUAL(ref.state. ...);
```

File: libvast/test/system/index.cpp (hunk 1 above)
Comment (mavam) [reply libvast/test/system/index.cpp]: Nope. Fixed.

File: libvast/test/system/index.cpp (hunk 2 above)
Comment (mavam) [reply libvast/test/system/index.cpp]: Exactly the code piece I was looking for but couldn't find when skimming the code base. Thanks! I've added this snippet as separate commit.

**Pull request merged**